src/parser/generated/regexp-tree.js

```diff
@@ -1104,19 +1105,68 @@ function GroupRefOrDecChar(text, textLoc) {
 /**
  * Unicode names.
  */
-const uRe = /^\\u[0-9a-fA-F]{4}/;
-const ucpRe = /^\\u\{[0-9a-fA-F]{1,}\}/;
+const uReStart = /^\\u[0-9a-fA-F]{4}/; // only matches start of string
+const ucpReStart = /^\\u\{[0-9a-fA-F]{1,}\}/; // only matches start of string
+const ucpReAnywhere = /\\u\{[0-9a-fA-F]{1,}\}/; // matches anywhere in string
```

Comment:
Is it a problem that this matches e.g. '\\\\u{2021}' (the even-odd problem)?

Reply:
I'm not certain that would be the case. `\\` isn't otherwise allowed in a group name, which has already been parsed prior to this regexp being used.